corehq/form_processor/models/forms.py

```diff
@@ -223,6 +226,25 @@ def set_archived_state(self, form, archive, user_id):
                            [case_ids, form_id, archive])
         form.state = self.model.ARCHIVED if archive else self.model.NORMAL
 
+    @classmethod
+    def publish_archive_action_to_kafka(cls, form, user_id, archive):
+        with cls._unfinished_archive(form, archive, user_id):
+            pass
+
+    @staticmethod
+    @contextmanager
+    def _unfinished_archive(form, archive, user_id, trigger_signals=True):
+        from ..change_publishers import publish_form_saved
+        with unfinished_archive(instance=form, user_id=user_id, archive=archive) as archive_stub:
+            yield archive_stub
+            is_sql = isinstance(form, XFormInstance)
+            if is_sql:
```

Comment:
these lines can be removed

Reply:
Thanks for catching that. I'll do it in the next PR.